Sonobuoy should validate regexp for focus/skip before the run

**What steps did you take and what happened:**
```
# /usr/local/bin/ginkgo --dryRun=true -v --focus="*" /usr/local/bin/e2e.test
I0228 20:03:18.361591     140 test_context.go:358] Using a temporary kubeconfig file from in-cluster config : /tmp/kubeconfig-676595395
Feb 28 20:03:18.361: INFO: The --provider flag is not set.  Treating as a conformance test.  Some tests may not be run.
I0228 20:03:18.361863     140 e2e.go:224] Starting e2e run "e3cfa9c2-3b93-11e9-a699-0242ac110005" on Ginkgo node 1
--- FAIL: TestE2E (0.02s)
panic: regexp: Compile(`*`): error parsing regexp: missing argument to repetition operator: `*` [recovered]
	panic: regexp: Compile(`*`): error parsing regexp: missing argument to repetition operator: `*`

goroutine 37 [running]:
testing.tRunner.func1(0xc0018a0800)
	/usr/local/go/src/testing/testing.go:792 +0x387
panic(0x398d2c0, 0xc0004e4020)
	/usr/local/go/src/runtime/panic.go:513 +0x1b9
regexp.MustCompile(0x7ffc15df6d8a, 0x1, 0x14)
	/usr/local/go/src/regexp/regexp.go:245 +0x171
k8s.io/kubernetes/vendor/github.com/onsi/ginkgo/internal/spec.(*Specs).applyRegExpFocusAndSkip(0xc0016b83e0, 0x44c000a, 0x14, 0x7ffc15df6d8a, 0x1, 0x0, 0x0)
	/workspace/anago-v1.13.0-rc.2.1+ddf47ac13c1a94/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/onsi/ginkgo/internal/spec/specs.go:88 +0x17b
k8s.io/kubernetes/vendor/github.com/onsi/ginkgo/internal/spec.(*Specs).ApplyFocus(0xc0016b83e0, 0x44c000a, 0x14, 0x7ffc15df6d8a, 0x1, 0x0, 0x0)
	/workspace/anago-v1.13.0-rc.2.1+ddf47ac13c1a94/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/onsi/ginkgo/internal/spec/specs.go:43 +0x75
k8s.io/kubernetes/vendor/github.com/onsi/ginkgo/internal/suite.(*Suite).generateSpecsIterator(0xc0000d6140, 0x44c000a, 0x14, 0x5c783e85, 0x1, 0x7ffc15df6d8a, 0x1, 0x0, 0x0, 0x0, ...)
	/workspace/anago-v1.13.0-rc.2.1+ddf47ac13c1a94/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/onsi/ginkgo/internal/suite/suite.go:83 +0x2bc
k8s.io/kubernetes/vendor/github.com/onsi/ginkgo/internal/suite.(*Suite).Run(0xc0000d6140, 0x7f1f794eeaf0, 0xc0018a0800, 0x44c000a, 0x14, 0xc0003cbc50, 0x1, 0x1, 0x4adb160, 0xc0000acde0, ...)
	/workspace/anago-v1.13.0-rc.2.1+ddf47ac13c1a94/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/onsi/ginkgo/internal/suite/suite.go:58 +0x180
k8s.io/kubernetes/vendor/github.com/onsi/ginkgo.RunSpecsWithCustomReporters(0x4a8fca0, 0xc0018a0800, 0x44c000a, 0x14, 0xc00007fe80, 0x1, 0x1, 0x0)
	/workspace/anago-v1.13.0-rc.2.1+ddf47ac13c1a94/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/onsi/ginkgo/ginkgo_dsl.go:218 +0x247
k8s.io/kubernetes/vendor/github.com/onsi/ginkgo.RunSpecsWithDefaultAndCustomReporters(0x4a8fca0, 0xc0018a0800, 0x44c000a, 0x14, 0x0, 0x0, 0x0, 0x4ed1cc0)
	/workspace/anago-v1.13.0-rc.2.1+ddf47ac13c1a94/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/onsi/ginkgo/ginkgo_dsl.go:206 +0x129
k8s.io/kubernetes/test/e2e.RunE2ETests(0xc0018a0800)
	_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/e2e.go:226 +0x1f6
k8s.io/kubernetes/test/e2e.TestE2E(0xc0018a0800)
	_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/e2e_test.go:77 +0x2b
testing.tRunner(0xc0018a0800, 0x4641300)
	/usr/local/go/src/testing/testing.go:827 +0xbf
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:878 +0x353

Ginkgo ran 1 suite in 503.595258ms
Test Suite Failed

```
minikube on mac 
minikube version
minikube version: v0.34.1

**What did you expect to happen:**
some sane message rather than crash

**Anything else you would like to add:**


**Environment:**

- Sonobuoy version: 
```
sonobuoy version
Sonobuoy Version: v0.13.0
MinimumKubeVersion: 1.11.0
MaximumKubeVersion: 1.13.99
GitSHA:
API Version check skipped due to missing `--kubeconfig` or other error
```
- Kubernetes version: (use `kubectl version`):
```
kubectl version
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.3", GitCommit:"721bfa751924da8d1680787490c54b9179b1fed0", GitTreeState:"clean", BuildDate:"2019-02-04T04:49:22Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.3", GitCommit:"721bfa751924da8d1680787490c54b9179b1fed0", GitTreeState:"clean", BuildDate:"2019-02-01T20:00:57Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
```
- Kubernetes installer & version:
minikube on mac

--- a/cmd/sonobuoy/app/args.go
+++ b/cmd/sonobuoy/app/args.go
@@ -19,6 +19,7 @@ package app
 import (
 	"fmt"
 	"io/ioutil"
+	"regexp"
 	"strings"
 
 	"github.com/heptio/sonobuoy/pkg/image"
@@ -160,6 +161,9 @@ func GetE2EConfig(mode ops.Mode, flags *pflag.FlagSet) (*ops.E2EConfig, error) {
 		if err != nil {
 			return nil, errors.Wrap(err, "couldn't retrieve focus flag")
 		}
+		if _, err := regexp.Compile(focus); err != nil {
+			return nil, errors.Wrap(err, "focus flag fails regexp validation")
+		}
 		cfg.Focus = focus
 	}
 
@@ -167,6 +171,9 @@ func GetE2EConfig(mode ops.Mode, flags *pflag.FlagSet) (*ops.E2EConfig, error) {
 		skip, err := flags.GetString(e2eSkipFlag)
 		if err != nil {
 			return nil, errors.Wrap(err, "couldn't retrieve skip flag")
+		}
+		if _, err := regexp.Compile(skip); err != nil {
+			return nil, errors.Wrap(err, "skip flag fails regexp validation")
 		}
 		cfg.Skip = skip
 	}

--- a/cmd/sonobuoy/app/args_test.go
+++ b/cmd/sonobuoy/app/args_test.go
@@ -1,0 +1,90 @@
+/*
+Copyright the Sonobuoy contributors 2019
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"testing"
+
+	"github.com/spf13/pflag"
+
+	ops "github.com/heptio/sonobuoy/pkg/client"
+)
+
+func TestGetE2EConfig(t *testing.T) {
+	testCases := []struct {
+		desc      string
+		mode      ops.Mode
+		flagArgs  []string
+		expect    *ops.E2EConfig
+		expectErr bool
+	}{
+		{
+			desc:     "Default",
+			mode:     "Conformance",
+			flagArgs: []string{},
+			expect: &ops.E2EConfig{
+				Focus:    `\[Conformance\]`,
+				Skip:     `Alpha|\[(Disruptive|Feature:[^\]]+|Flaky)\]`,
+				Parallel: "1",
+			},
+		}, {
+			desc:     "Flags settable",
+			mode:     "Conformance",
+			flagArgs: []string{"--e2e-focus=foo", "--e2e-skip=bar", "--e2e-parallel=2"},
+			expect: &ops.E2EConfig{
+				Focus:    `foo`,
+				Skip:     `bar`,
+				Parallel: "2",
+			},
+		}, {
+			desc:      "Focus regexp validated settable",
+			mode:      "Conformance",
+			flagArgs:  []string{"--e2e-focus=*"},
+			expectErr: true,
+		}, {
+			desc:      "Skip regexp validated settable",
+			mode:      "Conformance",
+			flagArgs:  []string{"--e2e-skip=*"},
+			expectErr: true,
+		},
+	}
+	for _, tC := range testCases {
+		t.Run(tC.desc, func(t *testing.T) {
+			fs := pflag.NewFlagSet("test", pflag.ExitOnError)
+			AddE2EConfigFlags(fs)
+			fs.Parse(tC.flagArgs)
+			cfg, err := GetE2EConfig(tC.mode, fs)
+			switch {
+			case err != nil && !tC.expectErr:
+				t.Fatalf("Expected no error but got %v", err)
+			case err == nil && tC.expectErr:
+				t.Fatalf("Expected error but got none")
+			}
+
+			switch {
+			case cfg == nil && tC.expect != nil:
+				t.Fatalf("Expected value %+v but got nil", tC.expect)
+			case cfg != nil && tC.expect == nil:
+				t.Fatalf("Expected nil value but got %+v", cfg)
+			case cfg == nil && tC.expect == nil:
+				return
+			case *tC.expect != *cfg:
+				t.Errorf("Expected config %+v but got %+v", tC.expect, cfg)
+			}
+		})
+	}
+}

